src/_pytest/fixtures.py

```diff
@@ -1184,31 +1181,6 @@ def pytest_fixture_setup(
     return result
 
 
-def wrap_function_to_error_out_if_called_directly(
-    function: FixtureFunction,
-    fixture_marker: FixtureFunctionMarker,
-) -> FixtureFunction:
-    """Wrap the given fixture function so we can raise an error about it being called directly,
-    instead of used as an argument in a test function."""
-    name = fixture_marker.name or function.__name__
-    message = (
-        f'Fixture "{name}" called directly. Fixtures are not meant to be called directly,\n'
-        "but are created automatically when test functions request them as parameters.\n"
-        "See https://docs.pytest.org/en/stable/explanation/fixtures.html for more information about fixtures, and\n"
-        "https://docs.pytest.org/en/stable/deprecations.html#calling-fixtures-directly about how to update your code."
-    )
-
-    @functools.wraps(function)
```

Comment:
Yes unfortunately the new result is this object instead of a wrapped function. I was not aware of this thanks for noticing it. Since the methods that are copied are not a lot shall I just assign them just like we do with `__name__`?

Reply:
I think the best thing would be to use `functools.update_wrapper` in `FixtureFunctionDefinition.__init__`, unless there is something preventing it.

Reply:
I did this the only problem was that when using 
`functools.update_wrapper(self, function)`
The FixtureFunctionDefinition will have a `__wrapped__` attribute which will be followed by `inspect.unwrap`. I removed this attribute so we don't accidentally unwrap the fixture without using `_get_wrapped_function` method.
If we don't remove this then we need to be careful to provide `stop` for `inspect.unwrap` when it's used.

Reply:
I was actually thinking that `__wrapper__` would be good to have, since `FixtureFunctionDefinition` is indeed a wrapper. What are the places where it causes an issue? (I can look it up myself a bit later, but in case you tried it already).

Reply:
Please see this commit that fixes the issue: https://github.com/pytest-dev/pytest/pull/12473/commits/475deb1ab6bb8d3509e9c8b3c8b6159f64f38f7d

I agree with doing this. Initially I was lazily binding methods, which led me to use `_get_wrapped_function`. Now with binding in the init we can rely on `inspect.unwrap` to get the underlying fixture. Less code for us. Thank you for the suggestion.